src/audio/module_adapter/module_adapter.c

```diff
@@ -828,19 +828,21 @@ static int module_adapter_simple_copy(struct comp_dev *dev)
 int module_adapter_copy(struct comp_dev *dev)
 {
 	struct processing_module *mod = comp_get_drvdata(dev);
-	struct module_data *md = &mod->priv;
+	struct module_data *md;
 	struct comp_buffer *source, *sink;
-	struct comp_buffer __sparse_cache *sink_c = NULL;
+	struct comp_buffer __sparse_cache *sink_c;
 	struct list_item *blist;
-	size_t size = MAX(mod->deep_buff_bytes, mod->period_bytes);
-	uint32_t min_free_frames = UINT_MAX;
-	int ret, i = 0;
+	size_t size;
+	uint32_t min_free_frames;
+	int ret, i;
```

Comment:
Hmm, not sure about this. First, I wonder if the compiler can already optimize this out? And if not, wouldn't it be easier to split module_adapter_copy into module_adapter_copy_complex() which is called if simple_copy is not used.

Reply:
this change make code looks strange and divided, although may save few cycles for stack init.
if really want, can use two functions with if else.

Reply:
Yes, the saving is tiny, need to re-think this, or close since saving is so small.